### PR TITLE
Cleanup more pointers

### PR DESCRIPTION
**Pull request by pcarruscag**
## Proposed Changes
Pointers are pure evil.

## PR Checklist
- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [X] My contribution is commented and consistent with SU2 style.
- [X] I have added a test case that demonstrates my contribution, if necessary.
- [X] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

**Comment by pcarruscag** [comment]
Why is it good C++ to use containers (instead of raw arrays):
- Book keeping: No need for verbose allocations / de-allocations.
- Information: The size is carried with the container, the type of data, the allocation strategy (e.g. alignment), etc.
- Correctness: A const method can change the elements of a raw array, because in such cases const does not qualify the data, only the address (pointer) of that data. On the other hand, a const method will imply const containers, on which only const methods can be called, and so on.
- Performance 1: A nested array (**), or nested container (vector<vector>) is NOT a matrix, it is a list of arrays, consequently there is more construction / destruction overhead (more allocations) uncertain performance at runtime, because inner arrays can end up arbitrarily far from each other, you end up needing copies to talk with e.g. blas or mpi...
- Versatility: Above all a container encapsulates data structure details (e.g. if something is a vector or list or map or row-major matrix or column-major matrix or sparse matrix) behind an interface. Which then lets you change the details separately from the code that uses the data. Whenever you pass an array by pointer you are locking yourself to ONE very particular organization of data in memory.
- Performance 2: This encapsulation is not insulation, it is completely transparent to the compiler because any half decent container type will be defined inline. Furthermore... If the compiler sees two pointers it has absolutely no idea if they are pointing to the same thing, or worse if they partially overlap. To legally optimize such code compilers need to introduce a ton of checks and fallback code, on the other hand, defining operations on containers which you know never to overlap provides a clean way to avoid compiler-paranoia.

**Comment by pcarruscag** [comment]
"But wait there's more"
- Debugging, profiling, logging: The access methods of containers can do bound-checking for debugging, they can be instrumented to measure how much data is used, or to log every imaginable type of information. Raw arrays will have you run to valgrind or address sanitizers and then good luck if the only example where you are able to reproduce a bug is too large and you need to wait one hour per test.

**Review by TobiKattmann — APPROVED** [review]
LGTM 💐 and thanks for the passionate plea for using containers instead of pointers. The jury withdraws for its final deliberations 🧑‍⚖️ . Jokes aside, it is really good to get some insights 
In order to nag about sth: I would appreciate a few more words in the PR description beyond bullying language features (Not sure what the pointer union has to say about that). Like using config->GetSolid_Wall instead of individual listings in several spaces; RMS_RES & BGS_RES vars now in containers for all(?) solvers (and all necessary work around that). Just some (more) highlights
(And have fun furthermore with `deleting 'em all` :) )

File: SU2_CFD/include/solvers/CSolver.hpp [hunk 1]
```diff
@@ -415,18 +421,16 @@ class CSolver {
...
-   * \brief Adds the maximal residual, this is useful for the convergence history.
-   * \param[in] val_var - Index of the variable.
-   * \param[in] val_residual - Value of the residual to store in the position <i>val_var</i>.
+   * \brief Set the RMS and MAX residual to zero.
    */
-  inline void AddRes_RMS(unsigned short val_var, su2double val_residual) { Residual_RMS[val_var] += val_residual; }
+  inline void SetResToZero() {
+    SU2_OMP_MASTER {
+      for (auto& r : Residual_RMS) r = 0;
+      for (auto& r : Residual_Max) r = 0;
```
[the remark this answers is not in the document]
Comment (pcarruscag) [reply SU2_CFD/include/solvers/CSolver.hpp:429]: There is a type in graph_toolbox.hpp called DummyGridColor with which one can write:
```C++
for (const auto iPoint : DummyGridColor<>(nPoint)) {
  // awesome code
}
```
Should have given it a better name like idk... Range :facepalm:
Anyway having that const auto iPoint is real useful when we go for them record breaking nested loops :+1:

File: SU2_CFD/include/solvers/CFVMFlowSolverBase.inl [hunk 2]
```diff
@@ -179,7 +170,7 @@ void CFVMFlowSolverBase<V, R>::Allocate(const CConfig& config) {
   SlidingState.resize(nMarker);
   SlidingStateNodes.resize(nMarker);
 
-  for (iMarker = 0; iMarker < nMarker; iMarker++) {
+  for (unsigned long iMarker = 0; iMarker < nMarker; iMarker++) {
```
[the remark this answers is not in the document]
Comment (pcarruscag) [reply SU2_CFD/include/solvers/CFVMFlowSolverBase.inl:173]: hmmmm ah yes, I cut and pasted it from the declaration at the top of the function :)

**Pull request merged**